File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/video/OutputResultsImpl.kt
package dev.hebei.camerax_android.video

import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.PigeonApiOutputResultsApi
import dev.hebei.camerax_android.common.api

class OutputResultsImpl(impl: CameraXImpl) : PigeonApiOutputResultsApi(impl) {
    override fun outputUri(pigeon_instance: androidx.camera.video.OutputResults): String? {
        return pigeon_instance.outputUri.api
    }
}

File: camerax_android/lib/src/video/output_results_impl.dart
import 'package:camerax_android/src/api.dart';
import 'package:camerax_android/src/common.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

final class OutputResultsImpl implements OutputResults {
  final OutputResultsProxyApi api;

  OutputResultsImpl.internal(this.api);

  @override
  Uri? get outputUri => api.outputUri?.uriImpl;
}

extension OutputResultsProxyApiX on OutputResultsProxyApi {
  OutputResults get impl => OutputResultsImpl.internal(this);
}
